src/main/java/com/contrastsecurity/maven/plugin/ContrastScanMojo.java

```diff
@@ -42,16 +52,44 @@
   private String projectId;
 
   /**
-   * File path to the Java artifact to upload for scanning. By default, uses the path to this
+   * File path of the Java artifact to upload for scanning. By default, uses the path to this
    * module's Maven artifact produced in the {@code package} phase.
    */
-  @Parameter(name = "artifactPath")
-  private Path artifact;
+  @Parameter private File artifactPath;
 
   /** A label to distinguish this scan from others in your project */
-  @Parameter(name = "label", defaultValue = "${project.version}")
+  @Parameter(defaultValue = "${project.version}")
   private String label;
 
+  /**
+   * When {@code true}, will wait for and retrieve scan results before completing the goal.
+   * Otherwise, will start a scan then complete the goal without waiting for Contrast Scan to
+   * complete.
+   */
+  @Parameter(defaultValue = "" + true)
+  private boolean waitForResults;
+
+  /** When {@code true}, will output a summary of the scan results to the console (build log). */
+  @Parameter(defaultValue = "" + true)
+  private boolean consoleOutput;
+
+  /**
+   * File path to where the scan results (in <a href="https://sarifweb.azurewebsites.net">SARIF</a>)
+   * will be written at the conclusion of the scan. Note: no results are written when {@link
+   * #waitForResults} is {@code false}.
+   */
+  @Parameter(
+      defaultValue =
+          "${project.build.directory}/contrast-scan-reports/contrast-scan-results.sarif.json")
+  private File outputPath;
+
+  /**
+   * Maximum time (in milliseconds) to wait for a Scan to complete. Scans that exceed this threshold
+   * fail this goal.
+   */
+  @Parameter(defaultValue = "" + 20 * 60 * 1000)
+  private long timeout;
```

Comment:
Contrast agent configuration uses `ms` suffix, so I'll continue that here

Reply:
🤔 Not calling `disconnect` isn't terrible, because the resource clean-up still happens in the `finally` block, but it's sloppy not to disconnect in this error case.

I think I have to introduce an extra condition to get it done, but it still looks clean to me. I'll fix this.